File: server/src/aif_types/embeddings.py
from typing import List
from pydantic import BaseModel
from sqlmodel import Field, SQLModel


class EmbeddingEntity(SQLModel, table=True):
    id: str = Field(primary_key=True)
    name: str
    vectorStoreProvider: str
    basemodel_uri: str

class CreateEmbeddingsRequest(BaseModel):
    input: str | List[str]
    name: str | None = None

class CreateOrUpdateEmbeddingsResponse(BaseModel):
    asset_id: str
    name: str

class ListEmbeddingsResponse(BaseModel):
    embeddings: List[EmbeddingEntity]

class UpdateEmbeddingMetadataRequest(BaseModel):
    name: str

File: packages/vscode-shared/src/types/api/embeddings.ts
import type { EmbeddingMetadata } from "../database/EmbeddingMetadata";

export type ListEmbeddingsResponse = {
	embeddings: EmbeddingMetadata[];
};

export type CreateOrUpdateEmbeddingsResponse = {
	id: string,
	name: string,
};

export type DeleteEmbeddingResponse = {
	id: string,
}
